lib/rules/require-scoped.ts

```diff
@@ -23,21 +26,174 @@ module.exports = {
+                        },
+                    },
+                },
+                additionalProperties: false,
+            },
+        ],
         type: "suggestion",
     },
     create(context: RuleContext) {
-        const always = context.options[0] !== "never"
         const styles = getStyleContexts(context).filter(isValidStyleContext)
         if (!styles.length) {
             return {}
         }
+
         const reporter = getCommentDirectivesReporter(context)
         const tokenStore = context.parserServices.getTemplateBodyTokenStore?.() as TokenStore
+        const { options } = context
+
+        if (options[0] === "error") {
+            const allows: AllowsOption = options[1]?.allows ?? ["scoped"]
+
+            /**
+             * Reports the given node.
+             * @param {ASTNode} node node to report
+             * @param {ASTNode} attribute type of style
+             */
+            function reportForbiddenStyle( // eslint-disable-line no-inner-declarations -- conditional report function
+                node: AST.VElement,
+                attribute: StyleTypes,
+            ) {
+                const singleAllow = allows.length === 1 && allows[0]
+                const forbiddenAttr = node.startTag.attributes.find(
+                    (attr) => attr.key.name === attribute,
+                )
+                const forbiddenAttrName = forbiddenAttr!.key.name as string
+
+                reporter.report({
+                    node: forbiddenAttr!,
+                    messageId: "forbiddenStyle",
+                    data: {
+                        attribute,
+                    },
+                    suggest: [
+                        singleAllow && singleAllow !== "plain"
+                            ? {
+                                  messageId: "change",
+                                  data: {
+                                      fromAttribute: forbiddenAttrName,
+                                      toAttribute: singleAllow,
+                                  },
+                                  fix(fixer: RuleFixer) {
+                                      return fixer.replaceText(
+                                          forbiddenAttr,
+                                          singleAllow,
+                                      )
+                                  },
+                              }
+                            : {
+                                  messageId: "remove",
+                                  data: {
+                                      attribute: forbiddenAttrName,
+                                  },
+                                  fix(fixer: RuleFixer) {
+                                      return fixer.remove(forbiddenAttr)
+                                  },
+                              },
+                    ],
+                })
+            }
+
+            /* eslint-disable no-inner-declarations -- conditional report function */
```

Comment:
@ota-meshi 

I had to temporarily disable the `no-inner-declarations` rule since I wasn't sure how you wanted me to organize the code here. These report functions are only used in the modern API, and require the reporter closure.

Do you have any preference?

Reply:
> temporarily disable the no-inner-declarations

That's okay :+1: